ocf_data_sampler/load/satellite.py

```diff
@@ -42,35 +60,160 @@ def get_single_sat_data(zarr_path: str) -> xr.Dataset:
+    da = da.assign_coords(channel=sorted(ds.data_vars))
+
+    # Standardize coordinate names and structure
+    da = da.rename({"time": "time_utc"})
+    check_time_unique_increasing(da.time_utc)
+    da = make_spatial_coords_increasing(da, x_coord="x_geostationary", y_coord="y_geostationary")
+    da = da.transpose("time_utc", "channel", "x_geostationary", "y_geostationary")
+
+    return da
+
+def _open_sat_data_zarr(
+    zarr_path: str | list[str], channels: Optional[List[str]] = None
+) -> xr.DataArray:
+    """Lazily opens a standard zarr store, potentially with multiple files."""
     if isinstance(zarr_path, list | tuple):
         ds = xr.combine_nested(
-            [get_single_sat_data(path) for path in zarr_path],
+            [_get_single_sat_data_zarr(path) for path in zarr_path],
             concat_dim="time",
             combine_attrs="override",
             join="override",
         )
     else:
-        ds = get_single_sat_data(zarr_path)
+        ds = _get_single_sat_data_zarr(zarr_path)
+
+    if channels:
+        ds = ds.sel(variable=channels)
 
     ds = ds.rename(
         {
             "variable": "channel",
             "time": "time_utc",
         },
     )
-
     check_time_unique_increasing(ds.time_utc)
     ds = make_spatial_coords_increasing(ds, x_coord="x_geostationary", y_coord="y_geostationary")
     ds = ds.transpose("time_utc", "channel", "x_geostationary", "y_geostationary")
 
-    # TODO: should we control the dtype of the DataArray?
     return get_xr_data_array_from_xr_dataset(ds)
+
+def open_sat_data(
+    zarr_path: str | list[str], channels: Optional[List[str]] = None
+) -> xr.DataArray:
+    """
+    Lazily opens satellite data, automatically detecting the format.
+
+    This function acts as a dispatcher based on Sol's feedback, routing to the 
+    appropriate loader based on the file path suffix. It supports:
+    1. Standard Zarr datasets (.zarr).
+    2. Ice Chunk repositories (.icechunk or .ic).
+
+    For Ice Chunk paths, the format should be:
+    `gs://<bucket>/<prefix>.icechunk[@<commit_sha>]`
+
+    Args:
+      zarr_path: Cloud URL or local path pattern, or list of these.
+      channels: A list of channel names to select from the dataset.
+    """
+    # Dispatch logic as requested by Sol
+    path_to_inspect = zarr_path[0] if isinstance(zarr_path, (list, tuple)) else zarr_path
+    
+    # Check the part of the path before any optional commit hash
+    main_path = path_to_inspect.split('@')[0]
+    
+    if main_path.endswith((".icechunk", ".ic")):
+        if isinstance(zarr_path, (list, tuple)):
+            raise NotImplementedError(
+                "Loading from multiple Ice Chunk repositories is not currently supported."
+            )
+        return _open_sat_data_icechunk(zarr_path, channels)
+    else:
+        logger.info(f"Opening satellite data from standard Zarr: {zarr_path}")
+        return _open_sat_data_zarr(zarr_path, channels)
```

Comment:
Apologies, as I think I previously said to put the logic in here; but I now think it is better suited in `_get_single_sat_data` as mentioned in [1] - that way the logic in the `open_sat_data` funtion can remain the same as it was and all the conditional opening logic can be in a single place in `_get_single_sat_data`.